lib/bolt/transport/remote.rb

```diff
@@ -45,9 +45,9 @@ def get_proxy(target)
       def run_task(target, task, arguments, options = {})
         proxy_target = get_proxy(target)
         transport = @executor.transport(proxy_target.protocol)
-        arguments = arguments.merge('_target' => target.to_h.reject { |_, v| v.nil? })
+        arguments = arguments.merge('_target' => target.options)
```

Comment:
I'm pretty sure this will prevent port etc from getting extracted from uris.

Reply:
Yeah, we're in a bind here. `target.to_h` (https://github.com/puppetlabs/bolt/blob/b4dcbcb2bfd21bc7b6a7374305a26b475d370e0e/lib/bolt/target.rb#L92-L102) overwrites `host` from the contents of `name`. This is especially awkward since we just recommended folks to use `host` as key in the transport schema (and released panos 1.0 with that) while https://github.com/puppetlabs/bolt/blob/b4dcbcb2bfd21bc7b6a7374305a26b475d370e0e/lib/bolt_server/transport_app.rb#L103 is a thing, and we can't just `s/hostname/host` to make that work for transports, that might not even have a `host` attribute

Another approach would be to say
```
target = [Bolt::Target.new(body['target']['host'] || body['target']['name'], opts)]
```
and ask the orchestrator to pass down the nodename/certname as `name`.

I wouldn't bother you with this if I weren't aware that all those options are unsatisfactory.

Reply:
This is just the way bolt works for now. You have to include the `host` in the name and there is no way to set it separately. We expect to fix this with version 2 of inventory but you'll need to fix it in ACE directly for now.

Are there modules that will accept but not require a `host` key in connection info. If thats not the case the second option seems like it will work.

Reply:
I expect we'll have inventory v2 functionality available as an opt-in within the next month so perhaps do something temporary now and fix it before ACE goes live.

Reply:
Cisco IOS uses `address` instead of host (but we can change that when porting to Transports), F5 will only provide and expect a single URI.

Longer term, cloud modules definitely have no host at all.

We'll continue exploring workarounds at the ACE API level then.